### PR TITLE
add except and notChar

### PR DESCRIPTION
**Pull request by andrevidela**
Hi, I needed those two functions in a project. @clayrat suggested they could fit here.

### CODE REVIEW

**Comment by gallais** [comment]
Thanks for the PR. I like the combinators, I'm not sure about the names.

Rather than `except` I would rather have `anyTokenBut`.
And `notChar` could be `anyCharBut` to keep the same naming scheme.

I think it would also be nice to have `noneOf : List (Tok p) -> All (Parser mn p (Tok p))`
as the converse of `anyOf` and the n-ary version of `anyTokenBut`.

**Comment by andrevidela** [comment]
@gallais I've implemented the changes you asked, could you take a look?

File: src/TParsec/Combinators.idr [hunk 1]
```diff
@@ -177,6 +177,14 @@ exacts : (Alternative mn, Monad mn, Inspect (Toks p) (Tok p), Eq (Tok p)) =>
...
 exacts ts = ands (map (\t => exact t) ts)
 
+anyTokenBut : (Alternative mn, Monad mn, Inspect (Toks p) (Tok p), Eq (Tok p)) =>
+              Tok p -> All (Parser mn p (Tok p))
+anyTokenBut t = guard (\t' => (t /= t')) anyTok
+
+noneOf : (Alternative mn, Monad mn, Inspect (Toks p) (Tok p), Eq (Tok p)) =>
+         List (Tok p) -> All (Parser mn p (Tok p))
+noneOf ts = guard (\t' => any (== t') ts) anyTok
+
```
Comment (gallais) [inline src/TParsec/Combinators.idr]: Don't you need to negate the condition here?

Comment (gallais) [reply src/TParsec/Combinators.idr]: Ok, I've pushed a fix. Waiting for the successful build.

**Pull request merged**